contracts/transmuter/src/alloyed_asset.rs

```diff
@@ -0,0 +1,69 @@
+use cosmwasm_std::{Addr, Coin, Deps, StdError, StdResult, Storage, Uint128};
+use cw_storage_plus::Item;
+use osmosis_std::types::cosmos::bank::v1beta1::BankQuerier;
+
+/// Alloyed asset represents the shares of the pool
+/// and since the pool is a 1:1 multi-asset pool, it act
+/// as a composite of the underlying assets and assume 1:1
+/// value to the underlying assets.
+pub struct AlloyedAsset<'a> {
+    alloyed_denom: Item<'a, String>,
+}
+
+impl<'a> AlloyedAsset<'a> {
+    pub const fn new(alloyed_denom_namespace: &'a str) -> Self {
+        Self {
+            alloyed_denom: Item::new(alloyed_denom_namespace),
+        }
+    }
+
+    /// get the alloyed denom
+    pub fn get_alloyed_denom(&self, store: &dyn Storage) -> StdResult<String> {
+        self.alloyed_denom.load(store)
```

Comment:
I generally like this pattern, but I'm thinking that it may lead to loads happening multiple times if the caller uses several get methods. 

Not release-blocker, but we can consider caching these (if the data is there return it, otherwise fetch)

Reply:
Nice idea! on it

Reply:
(just remember to reset the cache on sets! :p )